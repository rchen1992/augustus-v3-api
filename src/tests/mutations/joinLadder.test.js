const { testQuery } = require('@tests/utils');

const userId = '1';
const token = '3dV982mqxe4';

testQuery({
    id: 'Join a ladder',
    query: `
        mutation joinLadder($token: String!) {
            joinLadder(token: $token) {
                id
                ladderName
                userRating
                userRatingDelta
                createdAt
                inviteToken
            }
        }
    `,
    variables: {
        userId,
        token,
    },
    context: () => {
        return {
            currentUser: {
                user_id: userId,
            },
            services: {
                ladder: {
                    joinLadder() {
                        return {
                            ladder_id: 10,
                            ladder_name: 'Skyvu',
                            invite_token: token,
                            created_at: '2018-12-11 09:39:31',
                            updated_at: '2018-10-09 01:00:11',
                            ladder_user: {
                                ladder_id: 10,
                                user_id: userId,
                                rating: 1000,
                                rating_delta: 0,
                                created_at: '2018-12-11 09:39:31',
                                updated_at: '2018-10-09 01:00:11',
                            },
                        };
                    },
                },
            },
        };
    },
    expected: undefined,
});
