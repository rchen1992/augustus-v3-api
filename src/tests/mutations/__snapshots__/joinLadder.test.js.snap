// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query: Join a ladder 1`] = `
Object {
  "data": Object {
    "joinLadder": Object {
      "createdAt": "2018-12-11 09:39:31",
      "id": "10",
      "inviteToken": "3dV982mqxe4",
      "ladderName": "Skyvu",
      "userRating": 1000,
      "userRatingDelta": 0,
    },
  },
}
`;
